torch/csrc/jit/argument_spec.h

```diff
@@ -73,13 +73,20 @@ struct ArgumentSpec {
   }
 
   void addInput(const IValue& input, size_t& offset, bool with_grad) {
+    if (input.isTuple()) {
+      for (const IValue & elem : input.toTuple()->elements()) {
+        addInput(elem, offset, with_grad);
+      }
+      return;
+    }
```

Comment:
Why did that migrate above here? I specifically wanted to put the tensor case first, since it’s the most common by far, and I don’t want to increase overhead for w.g. CNNs that have hundreds of tensor parameters.

Reply:
This was just to remove some duplicated code and make sure the bounds check happened in every case since `args` was always being accessed. I've reverted it in the latest commit